addons/calendar/models/calendar_event.py

```diff
@@ -589,6 +590,14 @@ def action_mass_mailing(self):
             'context': compose_ctx,
         }
 
+    def action_join_meeting(self):
+        """ Method used when an existing user wants to join
+        """
+        self.ensure_one()
+        partner = self.env.user.partner_id
```

Comment:
Actually I was thinking the other way round: force giving partner_id(s) when calling it instead of forcing / fallbacking on current user's partner.